service/collector.go

```diff
@@ -243,6 +243,7 @@ func (col *Collector) shutdown(ctx context.Context) error {
 		errs = multierr.Append(errs, fmt.Errorf("failed to shutdown collector telemetry: %w", err))
 	}
 
+	col.service.telemetrySettings.Logger.Info("Shutdown complete.")
```

Comment:
This happens after the logger is closed.

Reply:
The log output in the PR description shows the "Shutdown complete.". How does this work if the logger is closed?

Reply:
It may be that the logger is not closed is just flushed, so the output is not deterministic?

Reply:
Oh I think I know why it works ;)
https://github.com/open-telemetry/opentelemetry-collector/blob/v0.55.0/service/service.go#L136

Perhaps I should first propose fixing that TODO and then put a "Logging finished" or similar message right before `Logger.Sync()` call.

Reply:
Oh I think I know why it works ;)
https://github.com/open-telemetry/opentelemetry-collector/blob/v0.55.0/service/service.go#L136

Perhaps I should first propose fixing that TODO and then put a "Logging finished" or similar message right before `Logger.Sync()` call.

Reply:
What is the reason we need to shutdown the Logger at all? Why shouldn't we keep it alive until the process is exited?

Reply:
Make sense. So, I think it is best to output "Service is shutdown" log message, followed by `Logger.Sync()` at the end of `service.Shutdown()`.

Reply:
>What is the reason we need to shutdown the Logger at all? Why shouldn't we keep it alive until the process is exited? 

This is not ok, because user can change the logger config, so we need to "close" it every time a config has changed (a.k.a service shutdown) not until the process ends.

Reply:
>This is not ok, because user can change the logger config, so we need to "close" it every time a config has changed (a.k.a service shutdown) not until the process ends.

I sill don't understand why we need to "close" it. I assume you are thinking about the service restart scenario, right? When restarting a new logger is created with potentially new config. Why is it a problem to keep the old logger around? It will be garbage collected when the new logger is created and assigned to the service's logger field. At least that's what I expect, maybe I am missing something.

Reply:
On the same note: "I also don't understand why is so important to have this log message?"

We definitely need to flush it to be deterministic, otherwise if after the restart we "end" too soon the previous log may miss things and may not be flushed. So why do we care to have this log and this debate? I think we should make "service" (instance) own the logger and its lifetime.